app/controllers/site_settings_controller.rb

```diff
@@ -0,0 +1,71 @@
+# Main site controller for user-editable site settings
+class SiteSettingsController < ApplicationController
+  include MainSiteHelper
+
+  before_action :authenticate_user!
+
+  def index
+    @settings =
+      if current_user_is_admin?
+        Setting.admin_settings
+      else
+        Setting.user_settings
+      end
+  end
+
+  def update
+    all_updated = update_settings( settings_params )
+    if all_updated.nil?
+      flash[ :notice ] = t( '.unchanged' )
+    elsif all_updated
+      flash[ :notice ] = t( '.success' )
+    end
+    redirect_to site_settings_path
+  end
+
+  private
+
+  def update_settings( params )
+    flags = initialise_flags
+    params&.each_key do |key|
+      next unless /value_(?<id>\d+)/ =~ key
+
+      flag  = update_value( id, params )
+      flags = update_flags( flag, flags )
+    end
+    return if flags[ :no_change ]
+
+    flags[ :no_errors ]
+  end
+
+  def update_value( id, params )
+    setting = Setting.find( id )
```

Comment:
Consider adding a further mitigation to prevent unauthorised updates, by scoping the `.find` call here:

```diff
-setting = Setting.find(id)
+settings = current_user_is_admin? Setting.admin_settings : Setting.user_settings
+setting = settings.find(id)
```

This scoping is applied in a few places, so consider extracting it into the `SettingPolicy` and making use of pundit's `policy_scope` method.